src/lib/blockchains/hype.coffee

```diff
@@ -0,0 +1,81 @@
+###
+
+dnschain
+http://dnschain.net
+
+Copyright (c) 2014 okTurtles Foundation
+
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this
+file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+###
+
+request = require 'superagent'
+
+module.exports = (dnschain) ->
+    # expose these into our namespace
+    for k of dnschain.globals
+        eval "var #{k} = dnschain.globals.#{k};"
+
+    BlockchainResolver = require('../blockchain.coffee')(dnschain)
+    ResolverStream  = require('../resolver-stream')(dnschain)
+
+    getAsync = Promise.promisify request.get
+
+    QTYPE_NAME = dns2.consts.QTYPE_TO_NAME
+    NAME_QTYPE = dns2.consts.NAME_TO_QTYPE
+    NAME_RCODE = dns2.consts.NAME_TO_RCODE
+    RCODE_NAME = dns2.consts.RCODE_TO_NAME
+    BLOCKS2SEC = 60
+
+    class NxtResolver extends BlockchainResolver
```

Comment:
If you were to create a new class (and this may not be necessary at all, see comments below), then this class name should be `HypeResolver`.